lib/nats.js

```diff
@@ -425,6 +435,9 @@ Client.prototype.sendConnect = function() {
     cs.user = this.user;
     cs.pass = this.pass;
   }
+  if (this.token !== undefined) {
+    cs.auth_token = this.token;
+  }
```

Comment:
Because of the varadic nature of option configuration it becomes possible to supply pass and token. In the case that both pass and token are provided likely we should fail the client connection and request that either password or token be provided.